### ISSUE TITLE
Notice: Undefined variable: serviceLocator

### ISSUE THREAD

**Issue by simonmacor** [zendframework/zend-mvc#148]
Hi,
The error is in Service/DiAbstractServiceFactoryFactory.php on line 32
I'm getting this error when I try create a service with a factory.

```
public function __invoke(ContainerInterface $container, $name, array $options = null)
    {
        $factory = new DiAbstractServiceFactory($container->get('Di'), DiAbstractServiceFactory::USE_SL_BEFORE_DI);

        if ($serviceLocator instanceof ServiceManager) {
            $serviceLocator->addAbstractFactory($factory, false);
        }

        return $factory;
    }
```

The problem seems to be present since the release 2.7.0

**Comment by michalbundyra** [comment]
@weierophinney could you please create branch `release-2.7` and then will be possible to create PR with the fix against this branch?

**Comment by weierophinney** [comment]
@webimpress Branch is now created, based on release-2.7.8. Ping me when you have a PR ready.

**Comment by weierophinney** [comment]
Fixed with #158.
